src/violation.rs

```diff
@@ -0,0 +1,9 @@
+use derive_builder::Builder;
+use tree_sitter::Node;
+
+#[derive(Builder)]
+#[builder(setter(into))]
+pub struct Violation<'a> {
+    pub message: String,
+    pub node: &'a Node<'a>,
```

Comment:
Right per other comment if `tree_sitter::Node` is `Copy` then it seems un-idiomatic to store a reference to it

Reply:
Yeah, and looks like it'd save you a little lifetime pain to not do it? (How is `Node` `Copy`, though? I'd have thought it held references to other things, which would require cloning.)

Reply:
I believe it's either a type alias of or a newtype wrapper around a raw pointer to the underlying `tree-sitter` C API `Node` type (and pointers are `Copy`)

Reply:
Ah right, and that works because the `Node` has an associated lifetime.